### ISSUE TITLE
Bug: Typescript plugin and parser are not working with flat config

### ISSUE THREAD

**Issue by igortas** [eslint/eslint#16706]
### Environment

Node version: v19.1.0
npm version: 8.19.3
Local ESLint version: v8.30.0 
Operating System:  Windows 11 WSL2

### What parser are you using?

@typescript-eslint/parser

### What did you do?

I have created new flat configuration, with `typescript` parser and `typescript` rules that overrides defaults `eslint` recommended rules.
On my project `eslint` was not working properly and was complaining mostly as un-def errors, for:
- import React
- interfaces
- types
- unused function props for function definitions in interfaces or types

Also, when I use the `typescript` plugin with short notation is not working, only works if I set the key as '@typescript/eslint': 
```
import typescript from '@typescript-eslint/eslint-plugin';
plugins: {
    typescript,
}
```

### What did you expect to happen?

I expect `typescript` parser to parse properly `.{ts, tsx}` files.

### What actually happened?

I'm getting errors as not recognizing the `typescript syntax`.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

```
import react from 'eslint-plugin-react';
import imports from 'eslint-plugin-import';
import a11y from 'eslint-plugin-jsx-a11y';
import typescript from '@typescript-eslint/eslint-plugin';
import globals from 'globals';

export default [
  'eslint:recommended',
  {
    files: ['src/**/*.ts', 'src/**/*.tsx'],
    languageOptions: {
      sourceType: 'module',
      ecmaVersion: 'latest',
      globals: {
        ...globals.browser,
      },
    },
    linterOptions: {
      noInlineConfig: true,
      reportUnusedDisableDirectives: true,
    },
    plugins: {
      react: react.configs.recommended,
      import: imports.configs.recommended,
      'jsx-a11y': a11y.configs.recommended,
      '@typescript-eslint': typescript,
    },
    rules: {
      'no-extra-parens': 'off',
      'no-unused-vars': 'off',
      'no-extra-semi': 'off',
      'no-unused-vars': 'off',
      'no-empty-function': 'off',

      'react/prop-types': 'off',
      'react/react-in-jsx-scope': 'off',

      '@typescript-eslint/no-extra-parens': 'off',
      '@typescript-eslint/no-unused-vars': 'error',
      '@typescript-eslint/adjacent-overload-signatures': 'error',
      '@typescript-eslint/ban-ts-comment': 'error',
      '@typescript-eslint/ban-types': 'error',
      '@typescript-eslint/no-array-constructor': 'error',
      '@typescript-eslint/no-empty-function': 'error',
      '@typescript-eslint/no-empty-interface': 'error',
      '@typescript-eslint/no-explicit-any': 'warn',
      '@typescript-eslint/no-extra-non-null-assertion': 'error',
      '@typescript-eslint/no-extra-semi': 'error',
      '@typescript-eslint/no-inferrable-types': 'error',
      '@typescript-eslint/no-loss-of-precision': 'error',
      '@typescript-eslint/no-misused-new': 'error',
      '@typescript-eslint/no-namespace': 'error',
      '@typescript-eslint/no-non-null-asserted-optional-chain': 'error',
      '@typescript-eslint/no-non-null-assertion': 'warn',
      '@typescript-eslint/no-this-alias': 'error',
      '@typescript-eslint/no-unnecessary-type-constraint': 'error',
      '@typescript-eslint/no-unused-vars': 'warn',
      '@typescript-eslint/no-var-requires': 'error',
      '@typescript-eslint/prefer-as-const': 'error',
      '@typescript-eslint/prefer-namespace-keyword': 'error',
      '@typescript-eslint/triple-slash-reference': 'error',
    },
  },
];
```
some of the errors like, no extra-parens or not detecting that is `typescript` interface:
```
  5:6  error  Parsing error: Unexpected token NonForwardedProps
  19:35  error  Parsing error: Unexpected token :
```

**Comment by aladdin-add** [comment]
>       '@typescript-eslint': typescript,

shouldn't it be `'@typescript-eslint': typescript.configs.recommended,`, just like others?

**Comment by igortas** [comment]
@aladdin-add It's not working that way, I've already tried. Not sure how needs to be set the new configuration with typescript. It lacks explanation in the docs.
